src/vos/vos_dtx.c

```diff
@@ -2208,11 +2206,62 @@ vos_dtx_commit(daos_handle_t coh, struct dtx_id dtis[], int count, bool rm_cos[]
 	return rc < 0 ? rc : committed;
 }
 
+int
+vos_dtx_abort_internal(struct vos_container *cont, struct vos_dtx_act_ent *dae, bool force)
+{
+	struct dtx_handle	*dth = dae->dae_dth;
+	struct umem_instance	*umm;
+	int			 rc;
+
+	umm = vos_cont2umm(cont);
+	rc = umem_tx_begin(umm, NULL);
+	if (rc != 0)
+		goto out;
+
+	if (dth != NULL) {
+		D_ASSERT(dth->dth_ent == dae || dth->dth_ent == NULL);
+		/* Not allow dtx_abort against solo DTX. */
+		D_ASSERT(!dth->dth_solo);
+		/* Set dth->dth_need_validation to notify the dth owner. */
+		dth->dth_need_validation = 1;
+	}
+
+	rc = dtx_rec_release(cont, dae, true);
+	if (rc == 0) {
+		dae->dae_aborting = 1;
+		rc = umem_tx_commit(umm);
+		D_ASSERTF(rc == 0, "local TX commit failure %d\n", rc);
+	} else {
+		rc = umem_tx_abort(umm, rc);
```

Comment:
just confirm do we need to set "dae->dae_aborting" flag also here?
umem_tx_abort() possibly yield similar as umem_tx_commit

Reply:
As my understand, the yield for local TX commit is necessary, because we cannot directly update WOL on SSD, instead, we put the commit request into the commit queue. But for local TX abort case, we do not need to update WOL since nothing on SSD has been updated if the local TX is failed. Since no yield, then do not need to set "dae_aborting"